ush/gefs_atmos_prep.sh

```diff
@@ -1,154 +1,176 @@
+  if [[ $tile != 6 ]]; then
+    OROG_FILES_TARGET_GRID=${OROG_FILES_TARGET_GRID}'","'
+  fi
+done
+export OROG_FILES_TARGET_GRID
+
 #############################################################
 # Execute the script
 $USHgfs/chgres_cube.sh
 export err=$?
 if [[ $err != 0 ]]; then
-	echo "FATAL ERROR in $(basename $BASH_SOURCE): chgres_cube failed!"
-	exit $err
+  echo "FATAL ERROR in $(basename $BASH_SOURCE): chgres_cube failed!"
+  exit $err
 fi
 #############################################################
 
-# Move files to the nwges directory
-for tile in tile1 tile2 tile3 tile4 tile5 tile6; do
-	mv ${DATA}/out.atm.${tile}.nc $OUTDIR/gfs_data.${tile}.nc
-done
-mv ${DATA}/gfs_ctrl.nc $OUTDIR/.
-
 touch ${OUTDIR}/chgres_atm.log  # recenter can start now
 
-if [[ $CONVERT_SFC == ".true." ]]; then
-	# Copy sfc files to the nwges directory for all members
-	for mem2 in $memberlist; do
-		INITDIR2=$GESOUT/init/$mem2
-		mkdir -p $INITDIR2
-		for tile in tile1 tile2 tile3 tile4 tile5 tile6; do
-			$NCP ${DATA}/out.sfc.${tile}.nc $INITDIR2/sfc_data.${tile}.nc
-		done
-	done
-fi
-
-# Copy control file to init
-$NCP $OUTDIR/gfs_ctrl.nc $INITDIR
-
 if [[ $SENDCOM == "YES" ]]; then
-	MODCOM=$(echo ${NET}_${COMPONENT} | tr '[a-z]' '[A-Z]')
-	DBNTYP=${MODCOM}_INIT	
-	COMDIR=$COMOUT/init/$mem
-	mkdir -p $COMDIR
-	$NCP $OUTDIR/gfs_ctrl.nc $COMDIR
-	if [[ $SENDDBN = YES ]];then
-		$DBNROOT/bin/dbn_alert MODEL $DBNTYP $job $COMDIR/gfs_ctrl.nc
-	fi
-	if [[ $mem == "c00" ]]; then
-		$NCP $OUTDIR/gfs_data*.nc $COMDIR
-		if [[ $SENDDBN = YES ]];then
-			for tile in tile1 tile2 tile3 tile4 tile5 tile6; do
-				$DBNROOT/bin/dbn_alert MODEL $DBNTYP $job $COMDIR/gfs_data.${tile}.nc
-			done
-		fi
-	fi
-	if [[ $CONVERT_SFC == ".true." ]]; then
-		for mem2 in $memberlist; do
-			COMDIR2=$COMOUT/init/$mem2
-			mkdir -p $COMDIR2
-			for tile in tile1 tile2 tile3 tile4 tile5 tile6; do
-				$NCP $GESOUT/init/$mem/sfc_data.${tile}.nc $COMDIR2
-				if [[ $SENDDBN = YES ]];then
-					$DBNROOT/bin/dbn_alert MODEL $DBNTYP $job $COMDIR2/sfc_data.${tile}.nc
-				fi
-			done
-		done
-	fi
+  MODCOM=$(echo ${NET}_${COMPONENT} | tr '[a-z]' '[A-Z]')
+  DBNTYP=${MODCOM}_INIT
+  COMDIR=${OUTDIR} #$COMOUT/$mem/atmos/INPUT #/init/$mem
+  mkdir -p $COMDIR
+  $NCP ${DATA}/gfs_ctrl.nc $COMDIR
+  if [[ $SENDDBN = YES ]];then
+    $DBNROOT/bin/dbn_alert MODEL $DBNTYP $job $COMDIR/gfs_ctrl.nc
+  fi
+
+  for tile in tile1 tile2 tile3 tile4 tile5 tile6; do
+    $NCP ${DATA}/out.atm.${tile}.nc $OUTDIR/gfs_data.${tile}.nc
+    if [[ $SENDDBN = YES ]];then
+      $DBNROOT/bin/dbn_alert MODEL $DBNTYP $job $COMDIR/gfs_data.${tile}.nc
+    fi
+  done
+
+  if [[ $CONVERT_SFC == ".true." ]]; then
+    for mem2 in $memberlist; do
+      YMD=${PDY} HH=${cyc} MEMDIR=${mem2} generate_com -x COM_ATMOS_INPUT
+      COMDIR2=${COM_ATMOS_INPUT} #${echo "${!COM_ATMOS_INPUT_TMPL}"} # $(echo "${!COM_ATMOS_INPUT_TMPL}" | envsubst) #${COM_ATMOS_INPUT} #$COMOUT/${mem2}/atmos/INPUT #init/$mem2
```

Comment:
```suggestion
YMD=${PDY} HH=${cyc} MEMDIR=${mem2} generate_com COMDIR2:COM_ATMOS_INPUT_TMPL
```
(It is only used locally, no need to export.)